### PR TITLE
OTL-2195: Fluentd disabled by default for installer scripts

### PR DESCRIPTION
**Pull request by jeffreyc-splunk**
**Requirements**

- [X] Content follows Splunk guidelines for style and formatting.
- [X] You are contributing original content.

**Describe the change**

Required when https://github.com/signalfx/splunk-otel-collector/pull/3369 is released, which will disable installation of fluentd by default.

### CODE REVIEW

File: gdi/opentelemetry/install-linux.rst [hunk 1]
```diff
@@ -31,7 +31,7 @@ The following Linux distributions and versions are supported:
 The installer script deploys and configures these elements:
 
 * The Splunk Distribution of OpenTelemetry Collector for Linux
-* Fluentd, using the td-agent. See :ref:`fluentd-receiver` for more information
+* Fluentd, using the td-agent (disabled by default). See :ref:`fluentd-receiver` for more information
```
Comment (theletterf) [inline gdi/opentelemetry/install-linux.rst:34]:
```suggestion
* Fluentd, using the td-agent. Turned off by default. See :ref:`fluentd-receiver` for more information
```

File: gdi/opentelemetry/install-linux.rst [hunk 2]
```diff
@@ -48,7 +48,7 @@ To install the package using the installer script, follow these steps:
    curl -sSL https://dl.signalfx.com/splunk-otel-collector.sh > /tmp/splunk-otel-collector.sh;
    sudo sh /tmp/splunk-otel-collector.sh --realm SPLUNK_REALM --memory SPLUNK_MEMORY_TOTAL_MIB -- SPLUNK_ACCESS_TOKEN
 
-.. note:: If you don't have a Log Observer entitlement or don't wish to collect logs for the target host, use the ``--without-fluentd`` option to skip the installation of Fluentd when installing the Collector.
+.. note:: If you have a Log Observer entitlement or wish to collect logs for the target host with Fluentd, use the ``--with-fluentd`` option to also install Fluentd when installing the Collector.
```
Comment (theletterf) [inline gdi/opentelemetry/install-linux.rst:51]: @jeffreyc-splunk Would you require Fluentd when sending logs for LO? Wouldn't filelog receiver also work?

File: gdi/opentelemetry/install-linux.rst [hunk 3]
```diff
@@ -95,10 +95,14 @@ To skip these steps and use configured repos on the target system that provide t
 Configure Fluentd
 ---------------------------------------
 
-.. note::
-   If you don't have a Log Observer entitlement or don't wish to collect logs for the target host, use the ``--without-fluentd`` option to skip the installation of Fluentd when installing the Collector.
+To install Fluentd for log collection (disabled by default), run the installer script with the ``--with-fluentd`` option. For example:
```
Comment (theletterf) [inline gdi/opentelemetry/install-linux.rst:98]:
```suggestion
Fluentd is turned off by default. To install Fluentd for log collection, run the installer script with the ``--with-fluentd`` option. For example:
```

File: gdi/opentelemetry/install-linux.rst [hunk 4]
```diff
@@ -95,10 +95,14 @@ To skip these steps and use configured repos on the target system that provide t
...
+
+.. code-block:: bash
+
+   curl -sSL https://dl.signalfx.com/splunk-otel-collector.sh > /tmp/splunk-otel-collector.sh && \
+   sudo sh /tmp/splunk-otel-collector.sh --with-fluentd --realm SPLUNK_REALM -- SPLUNK_ACCESS_TOKEN
 
-By default, the Fluentd service is installed and configured to forward log events with the ``@SPLUNK`` label to the package, which then sends these events to the HEC ingest endpoint determined by the ``--realm <SPLUNK_REALM>`` option. For example, ``https://ingest.<SPLUNK_REALM>.signalfx.com/v1/log``.
+When enabled, the Fluentd service is configured by default to collect and forward log events with the ``@SPLUNK`` label to the Collector, which then sends these events to the HEC ingest endpoint determined by the ``--realm <SPLUNK_REALM>`` option. For example, ``https://ingest.<SPLUNK_REALM>.signalfx.com/v1/log``.
```
Comment (theletterf) [inline gdi/opentelemetry/install-linux.rst:105]:
```suggestion
When turned on, the Fluentd service is configured by default to collect and forward log events with the ``@SPLUNK`` label to the Collector, which then sends these events to the HEC ingest endpoint determined by the ``--realm <SPLUNK_REALM>`` option. For example, ``https://ingest.<SPLUNK_REALM>.signalfx.com/v1/log``.
```

File: gdi/opentelemetry/install-windows.rst [hunk 5]
```diff
@@ -57,7 +57,7 @@ depending on the installation method:
 Installer script
 ==========================
 
-The installer script is available for Windows 64-bit environments, and deploys and configures the Splunk Distribution of OpenTelemetry Collector for Windows and Fluentd (using the td-agent).
+The installer script is available for Windows 64-bit environments, and deploys and configures the Splunk Distribution of OpenTelemetry Collector for Windows and Fluentd (using the td-agent, disabled by default).
```
Comment (theletterf) [inline gdi/opentelemetry/install-windows.rst:60]:
```suggestion
The installer script is available for Windows 64-bit environments, and deploys and configures the Splunk Distribution of OpenTelemetry Collector for Windows and Fluentd through the td-agent, which is deactivated by default.
```

File: gdi/opentelemetry/install-windows.rst [hunk 6]
```diff
@@ -99,7 +99,13 @@ To configure proxy settings to install and run the OpenTelemetry Collector, see
...
 -------------------------------------------
 
-By default, the installation configures the fluentd service to forward log events with the ``@SPLUNK`` label and
+If you have a Log Observer entitlement or wish to collect logs for the target host with Fluentd, use the ``with_fluentd = 1`` option to also install Fluentd when installing the Collector. For example:
+
+.. code-block:: PowerShell
+
+  & {Set-ExecutionPolicy Bypass -Scope Process -Force; $script = ((New-Object System.Net.WebClient).DownloadString('https://dl.signalfx.com/splunk-otel-collector.ps1')); $params = @{access_token = "SPLUNK_ACCESS_TOKEN"; realm = "SPLUNK_REALM"; with_fluentd = 1}; Invoke-Command -ScriptBlock ([scriptblock]::Create(". {$script} $(&{$args} @params)"))}
+
+When enabled, the Fluentd service is configured by default to collect and forward log events with the ``@SPLUNK`` label to the Collector, which then
```
Comment (theletterf) [inline gdi/opentelemetry/install-windows.rst:108]:
```suggestion
When activated, the Fluentd service is configured by default to collect and forward log events with the ``@SPLUNK`` label to the Collector, which then
```

**Review by theletterf — CHANGES_REQUESTED** [review]
Good work! Just some style nits.

Two general questions:

- Does it still make sense to promote Fluentd for now?
- Does --with-fluentd install the component or just enables the preinstalled component?

File: gdi/opentelemetry/install-linux.rst (hunk 2 above)
Comment (jeffreyc-splunk) [reply gdi/opentelemetry/install-linux.rst:51]: > @jeffreyc-splunk Would you require Fluentd when sending logs for LO? Wouldn't filelog receiver also work?

I just updated this line for the inverse.  I guess the filelog receiver would work, but we currently don't include it in our default collector configs.

File: gdi/opentelemetry/install-windows.rst (hunk 6 above)
Comment (jeffreyc-splunk) [reply gdi/opentelemetry/install-windows.rst:108]: Should we use `activated`/`deactivated`, or `turned on`/`turned off` as you suggested in the linux doc?

**Comment by jeffreyc-splunk** [comment]
> Good work! Just some style nits.
> 
> Two general questions:
> 
> * Does it still make sense to promote Fluentd for now?

Since we currently don't include the filelog receiver in our default configs, we still need to provide the option for log collection.  These changes simply require users to explicitly opt-in to fluentd.

> * Does --with-fluentd install the component or just enables the preinstalled component?

Nothing has changed with how fluentd is installed if the `--with-fluentd` option is specified.  The default for the installer scripts will be `--without-fluentd`, which will skip fluentd installation altogether.

**Review by theletterf — APPROVED** [review]
Approved! Could you let us know once the feature is up?

**Review by aurbiztondo-splunk — APPROVED** [review]

**Pull request merged**